SS2-Project/Assets/Starstorm2/ContentClasses/SS2Survivor.cs

```diff
@@ -8,22 +8,47 @@
 using System.Threading.Tasks;
 using UnityEngine;
 using RoR2.ContentManagement;
-
 namespace SS2
 {
     /// <summary>
     /// <inheritdoc cref="ISurvivorContentPiece"/>
     /// </summary>
-    public abstract class SS2Survivor : ISurvivorContentPiece
+    public abstract class SS2Survivor : ISurvivorContentPiece, IContentPackModifier
     {
-        public abstract SurvivorDef SurvivorDef { get; }
-        public abstract NullableRef<GameObject> MasterPrefab { get; }
+        public  SurvivorAssetCollection AssetCollection { get; private set; }
+        public SurvivorDef SurvivorDef { get; private set; }
+        public NullableRef<GameObject> MasterPrefab { get; private set; }
         CharacterBody IGameObjectContentPiece<CharacterBody>.Component => CharacterPrefab.GetComponent<CharacterBody>();
         GameObject IContentPiece<GameObject>.Asset => CharacterPrefab;
-        public abstract GameObject CharacterPrefab { get; }
+        public GameObject CharacterPrefab { get; private set; }
 
         public abstract void Initialize();
         public abstract bool IsAvailable(ContentPack contentPack);
-        public abstract IEnumerator LoadContentAsync();
+
+        public abstract SS2AssetRequest<SurvivorAssetCollection> AssetRequest { get; }
+
+        public IEnumerator LoadContentAsync()
+        {
+            SS2AssetRequest<SurvivorAssetCollection> request = AssetRequest;
+
+            request.StartLoad();
+            while (!request.IsComplete)
+                yield return null;
+
+            AssetCollection = request.Asset;
+
+            CharacterPrefab = AssetCollection.bodyPrefab;
+            MasterPrefab = AssetCollection.masterPrefab;
+            SurvivorDef = AssetCollection.survivorDef;
+
+            OnAssetCollectionLoaded(AssetCollection);
+        }
+
+        public virtual void OnAssetCollectionLoaded(AssetCollection assetCollection) { }
+
+        public void ModifyContentPack(ContentPack contentPack)
+        {
+            contentPack.AddContentFromAssetCollection(AssetCollection);
```

Comment:
I think this method in particular should be virtual, there might be a scenario where we *dont* want to add a specific type of asset (IE: UnlockableDefs for an unlockAll). There's a custom version of AddContentFromAssetCollection that accepts a predicate